### ISSUE TITLE
Custom module won't sync to certain minions

### ISSUE THREAD

**Issue by ns-phennessy** [saltstack/salt#24558]
I'm relatively new to salt; but I believe I have everything setup correctly.

I ran `salt '*' saltutil.sync_all` to add a custom module to my minions

This is the output I receive when trying to test my module: 
```
dev2.patcave.info:
    "ufw.status" is not available.
haproxy1.patcave.info:
    Status: inactive
mc1.patcave.info:
    Status: inactive
dev3.patcave.info:
    Status: inactive
saltmaster.patcave.info:
    Status: active
    
    To                         Action      From
    --                         ------      ----
    22                         ALLOW       Anywhere
    4505/tcp                   ALLOW       Anywhere
    4506/tcp                   ALLOW       Anywhere
    22 (v6)                    ALLOW       Anywhere (v6)
    4505/tcp (v6)              ALLOW       Anywhere (v6)
    4506/tcp (v6)              ALLOW       Anywhere (v6)
```

I investigated further on my `dev2` minion; and indeed found the correct version of my module in `/var/cache/salt/minion/`. There is absolutely no difference between my `dev2` and `dev3` boxes; they are identically configured, however it seemed to work fine on `dev3`

I deleted the copy it had on `dev2` and restarted the minion; then ran another sync, and it seemed to fix the issue. I could imagine this being a big issue if many minions fail to sync modules across a large cluster.

Not sure if it is a bug or some kind of caching issue? 

Thanks!

**Comment by jfindlay** [comment]
@ph7vc it seems to me that it is likely your `dev2` machine had a temporary connectivity issue which can happen in certain conditions.

**Comment by ph7vc** [comment]
I ran it multiple times and also checked it via the `test.ping` command. It would tell me the ufw.status is not available when I ran it, but test.ping came back fine. 

Though restarting the minion for some reason fixed it, so I really don't know why that would happen

**Comment by cachedout** [comment]
I have seen this happen. I don't think it's a connectivity issue. There's a race condition somwhere with module reloading and custom modules.

**Comment by cachedout** [comment]
I am almost certain we have this resolved in 2015.8. I haven't seen this in at least a year in a number of test scenarios. I'm going to close this as `Can't Reproduce`.
